clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1236,6 +1238,23 @@ DerivedArgList getLinkerArgs(ArrayRef<OffloadFile> Input,
                        Args.MakeArgString(Value));
   }
 
+  // Forward '-Xoffload-compiler' options to the appropriate backend.
+  for (StringRef Arg : Args.getAllArgValues(OPT_device_compiler_args_EQ)) {
+    auto [Triple, Value] = Arg.split('=');
+    llvm::Triple TT(Triple);
+    // If this isn't a recognized triple then it's an `arg=value` option.
+    if (TT.getArch() <= Triple::ArchType::UnknownArch ||
+        TT.getArch() > Triple::ArchType::LastArchType)
```

Comment:
I'd expect Triple to have an isvalid or operator bool or something instead of needing to do this range check

Reply:
I could probably use `Triple::parseArch(Str) == Triple::ArchType::UnknownArch)` instead, I think that's as close as we have.